lib/API.js

```diff
@@ -1046,109 +1056,120 @@ API.prototype._startJson = function(file, opts, action, pipe, cb) {
+      else
+        name = proc.name;
 
-    that.Client.getProcessIdByName(name, function(err, ids) {
-      if (err) {
-        Common.printError(err);
-        return next1();
-      }
-      if (!ids) return next1();
+      if (opts.only && opts.only != name)
+        return process.nextTick(next1);
 
-      async.eachLimit(ids, conf.CONCURRENT_ACTIONS, function(id, next2) {
-        var opts = {};
+      if (opts && opts.env)
+        new_env = Common.mergeEnvironmentVariables(proc, opts.env);
+      else
+        new_env = Common.mergeEnvironmentVariables(proc);
 
-        //stopProcessId could accept options to?
-        if (action == 'restartProcessId') {
-          opts = {id : id, env : new_env};
-        } else {
-          opts = id;
+      that.Client.getProcessIdByName(name, function(err, ids) {
+        if (err) {
+          Common.printError(err);
+          return next1();
         }
+        if (!ids) return next1();
 
-        that.Client.executeRemote(action, opts, function(err, res) {
-          ret_processes.push(res);
-          if (err) {
-            Common.printError(err);
-            return next2();
-          }
+        async.eachLimit(ids, conf.CONCURRENT_ACTIONS, function(id, next2) {
+          var opts = {};
 
+          //stopProcessId could accept options to?
           if (action == 'restartProcessId') {
-            that.Client.notifyGod('restart', id);
-          } else if (action == 'deleteProcessId') {
-            that.Client.notifyGod('delete', id);
-          } else if (action == 'stopProcessId') {
-            that.Client.notifyGod('stop', id);
+            opts = {id : id, env : new_env};
+          } else {
+            opts = id;
           }
 
-          Common.printOut(conf.PREFIX_MSG + '[%s](%d) \u2713', name, id);
-          return next2();
+          that.Client.executeRemote(action, opts, function(err, res) {
+            ret_processes.push(res);
+            if (err) {
+              Common.printError(err);
+              return next2();
+            }
+
+            if (action == 'restartProcessId') {
+              that.Client.notifyGod('restart', id);
+            } else if (action == 'deleteProcessId') {
+              that.Client.notifyGod('delete', id);
+            } else if (action == 'stopProcessId') {
+              that.Client.notifyGod('stop', id);
+            }
+
+            Common.printOut(conf.PREFIX_MSG + '[%s](%d) \u2713', name, id);
+            return next2();
+          });
+        }, function(err) {
+          return next1(null, ret_processes);
         });
-      }, function(err) {
-        return next1(null, ret_processes);
       });
+    }, function(err) {
+      if (cb) return cb(null, ret_processes);
+      else return that.speedList();
     });
-  }, function(err) {
-    if (cb) return cb(null, ret_processes);
-    else return that.speedList();
-  });
+  })
+
+  function getAppConf() {
+    return new Promise(function (resolve) {
```

Comment:
@soyuka Yes, but if I use this way, I must add `return Promise.resolve(xxx)` or `return Promise.resolve(xxx)` in every return statement, I think a function only have one return statement is more clear. Also, I don't understand why avoid `new Promise `, In my project, I almost use `new Promise` everywhere when I need to use promise, is it wrong to use `new Promise`?